### PR TITLE
fix bug some bug

### PR DESCRIPTION
**Pull request by davidwudv**
fix bug: memory leaks.
fix bug: crash by NULL pointer.
fix bug: 'upsync_interval' option does not take effect correct.

### CODE REVIEW

File: src/ngx_http_upsync_module.c [hunk 1]
```diff
@@ -2391,14 +2395,13 @@ ngx_http_upsync_add_timers(ngx_cycle_t *cycle)
     if (upsync_server == NULL) {
         return NGX_OK;
     }
-    upscf = upsync_server->upscf;
 
     ngx_log_debug1(NGX_LOG_DEBUG_HTTP, cycle->log, 0,
-                   "upsync_add_timers: shm_name: %V", upsync_server->pc.name);
+        "upsync_add_timers: shm_name: %V", "upsync_add_timers");
```
Comment (xiaokai-wang) [inline src/ngx_http_upsync_module.c]: "upsync_add_timers" that is the type of str, not the type of ngx_str_t, cannt use "%V".

Just be replaced by ' ngx_log_debug0(NGX_LOG_DEBUG_HTTP, cycle->log, 0, "upsync_add_timers"); ' in one line will be ok. thanks.

File: src/ngx_http_upsync_module.c [hunk 2]
```diff
@@ -2391,14 +2395,13 @@ ngx_http_upsync_add_timers(ngx_cycle_t *cycle)
...
 
     srandom(ngx_pid);
     for (i = 0; i < upsync_ctx->upstream_num; i++) {
-
+        upscf = upsync_server[i].upscf;
```
Comment (xiaokai-wang) [inline src/ngx_http_upsync_module.c]: the line is not logic relations with the lines below, moved to the line 2417 above(tmp = upscf->upsync_interval), that will be more better?
what do you think?

File: src/ngx_http_upsync_module.c (hunk 1 above)
Comment (davidwudv) [inline src/ngx_http_upsync_module.c]: @xiaokai-wang sorry

**Comment by davidwudv** [comment]
粗心大意了

**Pull request closed without merging**